src/workerd/util/sqlite.c++

```diff
@@ -139,27 +166,35 @@ SqliteDatabase::~SqliteDatabase() noexcept(false) {
   KJ_REQUIRE(err == SQLITE_OK, sqlite3_errstr(err)) { break; }
 }
 
-kj::Own<sqlite3_stmt> SqliteDatabase::prepareSql(sqlite3* db, kj::StringPtr sqlCode, uint prepFlags,
-                                                 Multi multi) {
+kj::Own<sqlite3_stmt> SqliteDatabase::prepareSql(
+    Regulator& regulator, kj::StringPtr sqlCode, uint prepFlags, Multi multi) {
+  // Set up the regulator that will be used for authorizer callbacks while preparing this
+  // statement.
+  KJ_ASSERT(currentRegulator == nullptr, "recursive prepareSql()?");
+  KJ_DEFER(currentRegulator = nullptr);
+  currentRegulator = regulator;
+
   for (;;) {
     sqlite3_stmt* result;
     const char* tail;
 
     SQLITE_CALL(sqlite3_prepare_v3(db, sqlCode.begin(), sqlCode.size(), prepFlags, &result, &tail));
-    KJ_REQUIRE(result != nullptr, "SQL code did not contain a statement", sqlCode);
+    SQLITE_REQUIRE(result != nullptr, "SQL code did not contain a statement.", sqlCode);
     auto ownResult = ownSqlite(result);
 
     while (*tail == ' ' || *tail == '\n') ++tail;
 
     switch (multi) {
       case SINGLE:
-        KJ_REQUIRE(tail == sqlCode.end(), "sqlite statement had leftover code", tail);
+        SQLITE_REQUIRE(tail == sqlCode.end(),
+            "A prepared SQL statement must contain only one statement.", tail);
         break;
 
       case MULTI:
         if (tail != sqlCode.end()) {
-          KJ_REQUIRE(sqlite3_bind_parameter_count(result) == 0,
-              "only the last statement in a multi-statement query can have parameters");
+          SQLITE_REQUIRE(sqlite3_bind_parameter_count(result) == 0,
+              "When executing multiple SQL statements in a single call, only the last statement "
+              "can have parameters.");
```

Comment:
I'm a bit confused here why `MULTI` exists at all? I suppose it doesn't hurt, but in D1 we've been splitting the queries up ahead of time and sending them in one-by-one, so we can provide per-query results/metrics.